lib/buffer.js

```diff
@@ -15,6 +15,49 @@ var poolSize, poolOffset, allocPool;
 
 
 binding.setupBufferJS(Buffer.prototype, bindingObj);
+
+const swap16n = Buffer.prototype.swap16;
+const swap32n = Buffer.prototype.swap32;
+
+Buffer.prototype.swap16 = function swap16() {
+  // For Buffer.length < 512, it's generally faster to
+  // do the swap in javascript. For larger buffers,
+  // dropping down to the native code is faster.
+  const len = this.length;
+  if (len < 512) {
+    if (len % 2 !== 0)
+      throw new RangeError('Buffer length must be a multiple of 16-bits');
+    for (var i = 0, n = 0; i < this.length; i += 2) {
```

Comment:
you already have `len` above... you might as well use it here too! :D 
(Same for `swap32`)